tests/tests/test_mtls.py

```diff
@@ -0,0 +1,315 @@
+    script = b"""\
+#! /bin/bash
+
+set -xe
+
+# Just give it a little bit of time
+sleep 6s
+
+# Successful update
+exit 0
+"""
+
+    with tempfile.NamedTemporaryFile(suffix="testdeployment") as tf:
+        tf.write(script)
+        tf.seek(0)
+        out = tf.read()
+        logger.info(f"Script: {out}")
+        script_path = tf.name
+
+        cmd = f"mender-artifact write module-image -T script -n {artifact_name} -t {device_type} -o {output_path} -f {script_path}"
+        logger.info(f"Executing command: {cmd}")
+        subprocess.check_call(cmd, shell=True)
+
+        return output_path
+
+
+class TestClientMTLSEnterprise:
+    @MenderTesting.fast
+    @pytest.mark.parametrize("algorithm", ["rsa", "ec256", "ed25519",])
+    def test_mtls_enterprise(self, setup_ent_mtls, algorithm):
+        tenant = setup_ent_mtls.tenant
+
+        # stop the api gateway
+        setup_ent_mtls.stop_api_gateway()
+
+        # start a new mender client
+        setup_ent_mtls.new_mtls_client("mender-client", tenant.tenant_token)
+
+        # stop the Mender client
+        setup_ent_mtls.device = device = MenderDevice(
+            setup_ent_mtls.get_mender_clients()[0]
+        )
+        device.ssh_is_opened()
+        device.run("systemctl stop mender-client")
+        tmpdir = tempfile.mkdtemp()
+        try:
+            # retrieve the original configuration file
+            output = device.run("cat /etc/mender/mender.conf")
+            config = json.loads(output)
+            # replace mender.conf with an mTLS enabled one
+            config["ServerURL"] = "https://mtls-ambassador:8080"
+            config["ServerCertificate"] = "/etc/mender/server.crt"
+            config["SkipVerify"] = True
+            config["HttpsClient"] = {
+                "Certificate": "/etc/mender/client.1.%s.crt" % algorithm,
+                "Key": "/etc/mender/client.1.%s.key" % algorithm,
+            }
+            del config["ArtifactVerifyKey"]
+            device.run("rm /data/mender/mender.conf")
+            mender_conf = os.path.join(tmpdir, "mender.conf")
+            with open(mender_conf, "w") as fd:
+                json.dump(config, fd)
+            device.put(
+                os.path.basename(mender_conf),
+                local_path=os.path.dirname(mender_conf),
+                remote_path="/etc/mender",
+            )
+        finally:
+            shutil.rmtree(tmpdir)
+
+        # upload the certificates
+        device.put(
+            "tenant.ca.crt",
+            local_path=os.path.join(
+                os.path.dirname(__file__),
+                "..",
+                "..",
+                "extra",
+                "mtls",
+                "certs",
+                "tenant-ca",
+            ),
+            remote_path="/etc/ssl/certs",
+        )
+        device.put(
+            "server.crt",
+            local_path=os.path.join(
+                os.path.dirname(__file__),
+                "..",
```

Comment:
Question:
Is this not a bit excessive? We're not testing on anything but Unix. So the separator is always going to be `/`. Is not hardcoding this a bit easier to read?

Reply:
Also, the `../../extra/mtls/certs` prefix seems to be used a lot of places.

Reply:
I prefer to stick to os.path.join, but I refactored it a bit to use base path variables.